src/sources/fluent/mod.rs

```diff
@@ -101,13 +101,17 @@ impl TcpSource for FluentSource {
         FluentDecoder::new()
     }
 
-    fn handle_events(&self, events: &mut [Event], host: Bytes) {
+    fn handle_events(&self, events: &mut [Event], host: Bytes, certificate_metadata: &Option<CertificateMetadata>) {
         for event in events {
             let log = event.as_mut_log();
 
             if !log.contains(log_schema().host_key()) {
                 log.insert(log_schema().host_key(), host.clone());
             }
+
+            if let Some(certificate_metadata) = certificate_metadata {
+                log.insert("certificate_metadata", certificate_metadata.to_string());
+            }
```

Comment:
In addition to @StephenWakely's comment, I'd also move this `log.insert` call up to `handle_stream` in `src/sources/util/tcp/mod.rs` since we want it to apply to all `tcp` based sources. That should save some duplication here.

Reply:
Just finished up an implementation of the above requested changes, let me know your thoughts!